### PR TITLE
fix(compiler): don't access view local variables nor pipes in host ex…

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/compiler/src/view_compiler/expression_converter.ts [hunk 1]
```diff
@@ -19,6 +20,23 @@ export interface NameResolver {
   createLiteralMap(values: Array<Array<string|o.Expression>>): o.Expression;
 }
```
Comment (vicb) [inline modules/@angular/compiler/src/view_compiler/expression_converter.ts]: one line comment

**Comment by vicb** [comment]
@tbosch please fix the tests

**Comment by alxhub** [comment]
@tbosch needs rebased, thanks!

**Comment by tbosch** [comment]
@alxhub Done

**Pull request merged**